### PR TITLE
[SNOW-1764119]  Eliminate unnecessary join for np.where with scalar for x 

### PR DESCRIPTION
**Pull request by sfc-gh-yzou**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1764119

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [x] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://docs.google.com/document/d/162d_i4zZ2AfcGRXojj0jByt8EUq-DrSHPPnTa4QvwbA/edit#bookmark=id.e82u4nekq80k)

3. Please describe how your code solves the related issue.

np.where with scalar x today requires 
1) temp table creation for broadcasting the scalar x to shape of cond
2) a join when doing pandas where

However, for such case, there should be no need of the extra temp table creation and join. In this change we removes the unnecessary temp table creation and join by using the indexing for scalar cast.

### CODE REVIEW

**Review by sfc-gh-evandenberg — APPROVED** [review]
LGTM

**Review by sfc-gh-mvashishtha — APPROVED** [review]
cool!

**Review by sfc-gh-jkew — APPROVED** [review]

File: src/snowflake/snowpark/modin/plugin/utils/numpy_to_pandas.py [hunk 1]
```diff
@@ -88,18 +88,33 @@ def is_same_shape(
...
         if is_scalar(x):
-            # broadcast scalar x to size of cond
-            object_shape = cond.shape
-            if len(object_shape) == 1:
-                df_scalar = pd.Series(x, index=range(object_shape[0]))
-            elif len(object_shape) == 2:
-                df_scalar = pd.DataFrame(
-                    x, index=range(object_shape[0]), columns=range(object_shape[1])
-                )
+            if cond.ndim == 1:
```
[the remark this answers is not in the document]
Comment (sfc-gh-jkew) [reply src/snowflake/snowpark/modin/plugin/utils/numpy_to_pandas.py:91]: got it - this is optimized out later

**Review by sfc-gh-helmeleegy — APPROVED** [review]
Nice work. Thanks, Yun!

**Pull request merged**